plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/hover/HoverProvider.java

```diff
@@ -105,4 +127,89 @@ private String renderContent(Hover hover) {
     }
     return Joiner.on("\n\n").join(contents);
   }
+
+  private void createHoverLinkListener(EditorPartPresenter activeEditor) {
+    final Window window = Elements.getWindow();
+    window.addEventListener(
+        Event.MOUSEDOWN,
+        evt -> {
+          Element anchorEle = (Element) evt.getTarget();
+
+          // Register the onClick and open only if the element is in a tooltip.
+          if (anchorEle.getOffsetParent() != null
+              && anchorEle.getOffsetParent().getClassList() != null
+              && anchorEle.getOffsetParent().getClassList().contains("textViewTooltipOnHover")
+              && anchorEle.hasAttribute("href")) {
+            String hrefContent = anchorEle.getAttribute("href");
+            Location uriLocation = getLocationFromURI(hrefContent);
+            anchorEle.setOnclick(
+                anchorEleClick -> {
+                  anchorEleClick.preventDefault();
+                  anchorEleClick.stopPropagation();
+                  if (hrefContent.startsWith("file:/") || hrefContent.startsWith("jdt:/")) {
+                    openFileInEditorHelper.openLocation(uriLocation);
+                  } else {
+                    BrowserUtils.openInNewTab(hrefContent);
+                  }
+                  ((TextEditor) activeEditor).getEditorWidget().hideTooltip();
+                });
+          }
+        });
+  }
+
+  /**
+   * Create a workable URI for opening files in the workspace
+   *
+   * @param uri The URI to fix
+   * @return A URI that can be opened by the editor
+   */
+  private String createWorkableURI(String uri) {
+    String fixedURI = uri;
+    if (uri.indexOf("#") != -1) {
+      fixedURI = fixedURI.substring(0, uri.lastIndexOf("#"));
+    }
+
+    // Because of the markdown2html library in jdt.ls file:///projects gets stripped to
+    // file:/projects when converting the markdown to html
+    if (uri.startsWith("file:/projects")) {
+      fixedURI = fixedURI.replace("file:/projects", "");
+    }
+
+    return fixedURI;
+  }
+
+  /**
+   * Gets the location from a URI that is in the form of URI#LineNumber
+   *
+   * @param uri The URI to get the location from
+   * @return Location object from the URI
+   */
+  private Location getLocationFromURI(String uri) {
+    String pattern = "\\d+$";
+    RegExp p = RegExp.compile(pattern);
+    MatchResult m = p.exec(uri);
+
+    Location uriLoc = new Location();
+    uriLoc.setUri(createWorkableURI(uri));
+
+    Position uriPos = new Position();
+
+    // We have line information from the URI
+    if (m != null && m.getGroupCount() > 0) {
+      String lastMatch = m.getGroup(m.getGroupCount() - 1);
+      Integer lineNumber = Integer.parseInt(lastMatch);
+
+      if (lineNumber > 0) {
+        uriPos.setLine(lineNumber - 1);
```

Comment:
Why do we do that?

Reply:
IIRC VSCode and Che have different behaviours on how they interpret the line they should jump to. In VSCode file://somefile#20 would link right to line 20 whereas with Che file://somefile#20 would link to line 21

Reply:
0-based vs. 1-based? But shouldn't we get the line number in the for LSP encodes it and shouldn't "OpenFileInEditorHelper" interpret that correctly?

Reply:
Currently [this](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/javadoc/JavaElementLinks.java#L180) is how its implemented in JDT.LS. 

> But shouldn't we get the line number in the for LSP encodes it 

I think you're actually correct. In the code I linked above the LSP encodes it correctly but for VSCode to actually jump to the correct line you have to add one because of their indexing.

> shouldn't "OpenFileInEditorHelper" interpret that correctly

I think the problem is that VSCode has their links being 1-based whereas in Che its correct which creates unfortunate consequences (like this one).